modules/entity_version_history/tests/src/Kernel/HistoryTabTest.php

```diff
@@ -0,0 +1,147 @@
+      'entity_version_history_test',
+    ]);
+
+    // Create a history tab setting for the corresponding entity type
+    // and bundle.
+    $history_storage = $this->container->get('entity_type.manager')->getStorage('entity_version_history_settings');
+    $history_storage->create([
+      'target_entity_type_id' => 'node',
+      'target_bundle' => 'first_bundle',
+      'target_field' => 'field_entity_version',
+    ])->save();
+
+    $this->container->get('entity_type.manager')->clearCachedDefinitions();
+  }
+
+  /**
+   * Tests the History routes.
+   */
+  public function testHistoryRoutes(): void {
+    $route_provider = $this->container->get('router.route_provider');
+    $local_task_manager = $this->container->get('plugin.manager.menu.local_task');
+
+    /** @var \Drupal\Core\Entity\EntityTypeInterface $definition */
+    foreach ($this->container->get('entity_type.manager')->getDefinitions() as $definition) {
+      if ($definition->id() === 'node') {
+        $this->assertTrue($definition->hasLinkTemplate('drupal:entity-version-history'));
+        $this->assertInstanceOf(Route::class, $route_provider->getRouteByName('entity.' . $definition->id() . '.history'));
+        $history_local_task = $local_task_manager->getDefinition('entity_version_history.entity.history:' . $definition->id());
+        $this->assertEquals('entity.' . $definition->id() . '.history', $history_local_task['route_name']);
+        $this->assertEquals('entity.' . $definition->id() . '.canonical', $history_local_task['base_route']);
+
+        continue;
+      }
+
+      $this->assertFalse($definition->hasLinkTemplate('drupal:entity-version-history'));
+
+      $exception = NULL;
+      try {
+        $route_provider->getRouteByName('entity.' . $definition->id() . '.history');
+      }
+      catch (\Exception $e) {
+        $exception = $e;
+      }
+      $this->assertInstanceOf(RouteNotFoundException::class, $exception);
+    }
+  }
+
+  /**
+   * Tests access to the entity version history page.
+   */
+  public function testHistoryPageAccess(): void {
+    // Define the anonymous user first. User id with 0 has to exist in order
+    // to avoid "ContextException: The 'entity:user' context is required and
+    // not present" error.
+    User::create([
+      'name' => 'Anonymous',
+      'uid' => 0,
+    ])->save();
+
+    $entity_type_manager = $this->container->get('entity_type.manager');
+
+    $node = $entity_type_manager->getStorage('node')->create([
+      'type' => 'first_bundle',
+      'title' => 'My node',
+    ]);
+    $node->save();
+
+    $user_with_permission = $this->createUser(['access entity version history']);
+    $user_without_permission = $this->createUser();
+
+    $history_url = Url::fromRoute('entity.node.history', [
+      'node' => $node->id(),
+    ]);
+
+    // Assert that we can't access the history page when no permissions are
+    // assigned.
+    $this->assertFalse($history_url->access($user_without_permission));
```

Comment:
Instead of using the Url::access(), use the AccessManager and its `checkNamedRoute` in order to get it to return the access result object. This is so you can assert the caches on that as well as the reason for not having access.